resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/kubernetes/KubernetesClusterSchedulerBackend.scala

```diff
@@ -368,14 +364,11 @@ private[spark] class KubernetesClusterSchedulerBackend(
    *         locality if an executor launches on the cluster node.
    */
   private def getNodesWithLocalTaskCounts() : Map[String, Int] = {
-    val executorPodsWithIPs = EXECUTOR_PODS_BY_IPS_LOCK.synchronized {
-      executorPodsByIPs.values.toList  // toList makes a defensive copy.
-    }
     val nodeToLocalTaskCount = mutable.Map[String, Int]() ++
       KubernetesClusterSchedulerBackend.this.synchronized {
         hostToLocalTaskCount
       }
-    for (pod <- executorPodsWithIPs) {
+    for ((_, pod) <- executorPodsByIPs) {
```

Comment:
I think the concern from @mccheah is valid if we care about the consistency of the map throughout the entire iterating. Iterating through a `ConcurrentHashMap` may or may not reflect changes made to the map after the iterator is created, although iteration is thread-safe and won't throw the `ConcurrentModificationException`. In this particular case, it means we may or may not lose changes made while the map is being iterated through. Also it is mentioned in the Javadoc that "iterators are designed to be used by only one thread at a time."